satpy/readers/pmw_channels_definitions.py

```diff
@@ -0,0 +1,415 @@
+                right_side_dist = abs(value - right_right)
+                return min(left_side_dist, right_side_dist)
+        else:
+            return np.inf
+
+    @classmethod
+    def convert(cls, frq):
+        """Convert `frq` to this type if possible."""
+        if isinstance(frq, dict):
+            return cls(**frq)
+        return frq
+
+
+class FrequencyDoubleSideBandBase(NamedTuple):
+    """Base class for a frequency double side band.
+
+    Frequency Double Side Band is supposed to describe the special type of bands
+    commonly used in humidty sounding from Passive Microwave Sensors. When the
+    absorption band being observed is symmetrical it is advantageous (giving
+    better NeDT) to sense in a band both right and left of the central
+    absorption frequency.
+
+    This is needed because of this bug: https://bugs.python.org/issue41629
+
+    """
+
+    central: float
+    side: float
+    bandwidth: float
+    unit: str = "GHz"
+
+
+class FrequencyDoubleSideBand(FrequencyDoubleSideBandBase):
+    """The frequency double side band class.
+
+    The elements of the double-side-band type frequency band are the central
+    frquency, the relative side band frequency (relative to the center - left
+    and right) and their bandwidths, and optionally a unit (defaults to
+    GHz). No clever unit conversion is done here, it's just used for checking
+    that two ranges are comparable.
+
+    Frequency Double Side Band is supposed to describe the special type of bands
+    commonly used in humidty sounding from Passive Microwave Sensors. When the
+    absorption band being observed is symmetrical it is advantageous (giving
+    better NeDT) to sense in a band both right and left of the central
+    absorption frequency.
+
+    """
+
+    def __eq__(self, other):
+        """Return if two channel frequencies are equal.
+
+        Args:
+            other (tuple or scalar): (central frq, side band frq and band width frq) or scalar frq
+
+        Return:
+            True if other is a scalar and min <= other <= max, or if other is
+            a tuple equal to self, False otherwise.
+
+        """
+        if other is None:
+            return False
+        if isinstance(other, numbers.Number):
+            return other in self
+        if isinstance(other, (tuple, list)) and len(other) == 3:
+            return other in self
+        return super().__eq__(other)
+
+    def __ne__(self, other):
+        """Return the opposite of `__eq__`."""
+        return not self == other
+
+    def __lt__(self, other):
+        """Compare to another frequency."""
+        if other is None:
+            return False
+        return super().__lt__(other)
+
+    def __gt__(self, other):
+        """Compare to another frequency."""
+        if other is None:
+            return True
+        return super().__gt__(other)
+
+    def __hash__(self):
+        """Hash this tuple."""
+        return tuple.__hash__(self)
```

Comment:
There seems to be some code repetition here.  Lines 100-108 seem to be the same.  Can this be factored out to a common base class?